### PR TITLE
ts: Write quotes as entities

### PR DESCRIPTION
**Pull request by dg0yt**
Removes final format difference with files saved from Qt Linguist.

### CODE REVIEW

**Comment by dg0yt** [comment]
This might deserve some more testing. But I only use hosted weblate...
Unfortunately, this slows down serialize significantly (tested with 1000 iterations in test_backnforth).

**Comment by dg0yt** [comment]
Now completely without regexps. The slowdow is still noticable, but no longer by factor 10...

File: translate/storage/ts2.py [hunk 1]
```diff
@@ -488,6 +489,20 @@ def serialize(self, out):
...
             e.text = ""
+        # For conformance with Qt output, write XML declaration with double quotes
         out.write(str('<?xml version="1.0" encoding="utf-8"?>\n').encode('utf-8'))
-        out.write(etree.tostring(root, doctype=doctype, pretty_print=True,
-                                 xml_declaration=None, encoding='utf-8'))
+        # For conformance with Qt output, post-process etree.tostring output,
+        # replacing ' with &apos; and " with &quot; in text elements
+        treestring = etree.tostring(root, doctype=doctype, pretty_print=True,
+                                    xml_declaration=False, encoding='utf-8')
+        treestring = codecs.decode(treestring, 'utf-8')
```
Comment (nijel) [inline translate/storage/ts2.py]: You should use encoding='unicode' for etree.tostring(), that saves need for the additional utf-8 decoding.

Comment (dg0yt) [inline translate/storage/ts2.py]: Done.

**Comment by dg0yt** [comment]
Ping. I would like to remove nasty git noise workarounds from our project's build system.

**Comment by unho** [comment]
As you said this brings a noticeable performance drop. Can you explore using regexes to speed it up?

**Comment by dg0yt** [comment]
Hm. I tried regexes first and it was slower by one order of magnitude.

File: translate/storage/ts2.py [hunk 2]
```diff
@@ -488,6 +488,19 @@ def serialize(self, out):
...
         for e in root.xpath("//*[not(./node()) and not(text()) and not(name() = 'location')]"):
             e.text = ""
+        # For conformance with Qt output, write XML declaration with double quotes
         out.write(str('<?xml version="1.0" encoding="utf-8"?>\n').encode('utf-8'))
-        out.write(etree.tostring(root, doctype=doctype, pretty_print=True,
-                                 xml_declaration=None, encoding='utf-8'))
+        # For conformance with Qt output, post-process etree.tostring output,
+        # replacing ' with &apos; and " with &quot; in text elements
+        treestring = etree.tostring(root, doctype=doctype, pretty_print=True,
+                                    xml_declaration=False, encoding='unicode')
```
Comment (dg0yt) [inline translate/storage/ts2.py]: Maybe one could do `encode('utf-8')` the whole string here, rather calling `encode` many times in the loop. But I'm not a python expert. I have hardly any knowledge about dealing with encoding of python strings.

Comment (nijel) [inline translate/storage/ts2.py]: Indeed it can be done simpler, see https://github.com/translate/translate/pull/3528

**Comment by unho** [comment]
Closing in favor of https://github.com/translate/translate/pull/3528

**Pull request closed without merging**